Common/src/main/java/com/yungnickyoung/minecraft/ribbits/mixin/mixins/client/ChannelMixin.java

```diff
@@ -0,0 +1,72 @@
+package com.yungnickyoung.minecraft.ribbits.mixin.mixins.client;
+
+import com.mojang.blaze3d.audio.Channel;
+import com.mojang.blaze3d.audio.SoundBuffer;
+import com.yungnickyoung.minecraft.ribbits.mixin.interfaces.client.IChannelDuck;
+import com.yungnickyoung.minecraft.ribbits.client.sound.RibbitInstrumentSoundInstance;
+import com.yungnickyoung.minecraft.ribbits.mixin.mixins.client.accessor.SoundBufferAccessor;
+import net.minecraft.client.resources.sounds.SoundInstance;
+import org.lwjgl.BufferUtils;
+import org.lwjgl.openal.AL10;
+import org.lwjgl.openal.AL11;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.nio.IntBuffer;
+import java.util.OptionalInt;
+
+@Mixin(Channel.class)
+public class ChannelMixin implements IChannelDuck {
+
+    @Shadow @Final private int source;
+
+    @Override
+    public void attachStaticBufferWithByteOffset(SoundInstance instance, SoundBuffer soundBuffer, int sourceId) {
```

Comment:
Need documentation on both of these attach methods, explaning what they do and when each of them is used